### ISSUE TITLE
Can create Server without IP address, network MTU, Interface Name

### ISSUE THREAD

**Issue by dpham692** [apache/trafficcontrol#4787]
## I'm submitting a ...

- [x] bug report
- [ ] new feature / enhancement request
- [ ] improvement request (usability, performance, tech debt, etc.)
- [ ] other 

## Traffic Control components affected ...

- [ ] CDN in a Box
- [ ] Documentation
- [ ] Grove
- [ ] Traffic Control Client
- [ ] Traffic Monitor
- [x] Traffic Ops
- [ ] Traffic Ops ORT
- [ ] Traffic Portal
- [ ] Traffic Router
- [ ] Traffic Stats
- [ ] Traffic Vault
- [ ] unknown

## Current behavior:

Right now, if user use `https://{{TO_BASE_URL}}/api/2.0/servers`, they can create a Server without IP address, network MTU, and Interface Name.
For example use this payload :
```
{
        "cachegroupId": 6,
        "cdnId": 1,
        "domainName": "infra.ciab.test",
        "hostName": "test",
        "httpsPort": 443,
        "iloIpAddress": "",
        "iloIpGateway": "",
        "iloIpNetmask": "",
        "iloPassword": "",
        "iloUsername": "",
        "interfaceMtu": 1500,
        "interfaceName": "eth0",
        "ip6Address": "::1",
        "ip6Gateway": "::2",
        "ipAddress": "0.0.0.1",
        "ipGateway": "0.0.0.2",
        "ipNetmask": "255.255.255.0",
        "mgmtIpAddress": "",
        "mgmtIpGateway": "",
        "mgmtIpNetmask": "",
        "offlineReason": "",
        "physLocationId": 1,
        "profileId": 3,
        "routerHostName": "",
        "routerPortName": "",
        "statusId": 3,
        "tcpPort": 80,
        "typeId": 12,
        "updPending": false,
        "ipIsService": true,
        "ip6IsService": true
}
```
It will response success 201 Created. Then go back to TP and go to the Servers just created. TP IP Address, network MTU, Interface Name will be blank.
## Expected / new behavior:

The fields should be blank.

**Comment by rawlinp** [comment]
> 201 Created

That seems bad. Changing the http status code can break things that expect it to return a 200 -- which is what it did before. Is that right @ocket8888?

**Comment by ocket8888** [comment]
Yeah, that's true. From the API guidelines' section on 201 Created:

> _"This MUST be the response status code of POST requests that create a new object or objects on the server..."_

It's tested and working with the Go client, Python client, and Traffic Portal.

**Comment by rawlinp** [comment]
Oh, you tested the Python client? It is definitely broken by this change:
```
>>> a, _ = TO.create_server(data=s)
ERROR:trafficops.restapi:restapi.py:_do_operation:424: POST request to RESTful API at [http://localhost:30003/api/2.0/servers] expected status(s) (200, 204); failed: 201 Created
```
I get that the API guidelines would like to move to 201s, but you can't do that without potentially breaking clients that depend on the previously-expected 200. Just adding 201 to the expected statuses without fixing this in the API is not good enough, unfortunately.

**Comment by ocket8888** [comment]
Yeah I guess testing the actual create_server method might be important. I actually only tested `request`, thought that was sufficient, since every other method just goes through that. Clearly not. I'll definitely fix that.

Anything that treats 201 as a failure is in violation of the HTTP spec, and needs to be fixed, IMO.

**Comment by rawlinp** [comment]
Sure, you can fix the existing python client to not treat 201 as failure, which fixes the client going forward, but everyone using prior versions of the client will be broken when the API starts returning 201s instead of 200s. IMO if you want to return a 201 it should be done v3 only, since v1 and v2 returned 200 and client behavior has been built around that.

**Comment by ocket8888** [comment]
> _"IMO if you want to return a 201 it should be done in v3 only, since v1 and v2 returned 200 and client behavior has been built around that"_

Actually, I agree. Looking back, I really shouldn't have changed it in v1 and v2. I'll fix that too.

**Comment by mitchell852** [comment]
@dpham692 @ocket8888 is this still an issue? i know a lot has been changing with servers in master and there were a few bumps but maybe this has been resolved by now?

**Comment by ocket8888** [comment]
No, I actually don't think this was ever an issue. It's just that before TP was updated to use interfaces, creating a server would result in those fields showing as blank in TP.

**Comment by mitchell852** [comment]
@ocket8888 so can close?

**Comment by ocket8888** [comment]
Well I'd like the guy who opened it to confirm it's not an issue.
